tests/unit/__snapshots__/documentation.spec.ts.snap

```diff
@@ -3763,6 +3763,181 @@ servers:
+"openapi: 3.1.0
+info:
+  title: Testing headers params
+  version: 3.4.5
+paths:
+  /v1/test:
+    post:
+      operationId: PostV1Test
+      parameters:
+        - name: id
+          in: query
+          required: true
+          description: POST /v1/test Parameter
+          schema:
+            type: string
+        - name: x-request-id
+          in: header
+          required: true
+          description: POST /v1/test Parameter
+          schema:
+            type: string
+      requestBody:
+        description: POST /v1/test Request body
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                id:
+                  type: string
+                x-request-id:
+                  type: string
+              required:
+                - id
+                - x-request-id
+      responses:
+        "200":
+          description: POST /v1/test Positive response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: string
+                    const: success
+                  data:
+                    type: object
+                required:
+                  - status
+                  - data
+        "400":
+          description: POST /v1/test Negative response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: string
+                    const: error
+                  error:
+                    type: object
+                    properties:
+                      message:
+                        type: string
+                    required:
+                      - message
+                required:
+                  - status
+                  - error
+              examples:
+                example1:
+                  value:
+                    status: error
+                    error:
+                      message: Sample error message
+components:
+  schemas: {}
+  responses: {}
+  parameters: {}
+  examples: {}
+  requestBodies: {}
+  headers: {}
+  securitySchemes: {}
+  links: {}
+  callbacks: {}
+tags: []
+servers:
+  - url: https://example.com
+"
+`;
+
+exports[`Documentation > Feature 1180: Headers opt-in params > should describe x- inputs as header params in put request 1`] = `
+"openapi: 3.1.0
+info:
+  title: Testing headers params
+  version: 3.4.5
+paths:
+  /v1/test:
+    put:
+      operationId: PutV1Test
+      requestBody:
+        description: PUT /v1/test Request body
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                id:
+                  type: string
+                x-request-id:
+                  type: string
```

Comment:
this is also wrong. Moreover, parameters are missing because "query" is disabled.
Additional case.

Reply:
missing header param fixed in 4845123f, but needs a refactoring

Reply:
fixed